tests/purs/failing/OverlappingUnnamedInstances.purs

```diff
@@ -0,0 +1,17 @@
+-- @shouldFailWith OverlappingInstances
+module Main where
+
+class Test a where
+  test :: a -> a
+
+instance Test a where
+  test x = x
+
+instance Test Int where
+  test _ = 0
+
+-- The OverlappingInstances instances error only arises when there are two
+-- choices for a dictionary, not when the instances are defined. So without
+-- `value` this module would not raise a warning.
```

Comment:
```suggestion
-- `value` this module would not raise a error.
```

Reply:
@mhmdanas This is from #4096, so maybe report it there?

Reply:
Oh, sorry. I forgot that this was not based on `master` :D.